### PR TITLE
Working version of matrix integration pfs

### PR DESCRIPTION
**Pull request by Dominik1999**

### CODE REVIEW

File: src/pathfinding_service/cli.py [hunk 1]
```diff
@@ -150,6 +176,8 @@ def main(
             registry_address=registry_address,
             sync_start_block=start_block,
             required_confirmations=confirmations,
+            private_key=private_key,
+            poll_interval=10,
```
Comment (palango) [inline src/pathfinding_service/cli.py]: I'd like to not define this here or use a default

File: src/pathfinding_service/service.py [hunk 2]
```diff
@@ -13,13 +13,17 @@
     create_channel_event_topics,
     create_registry_event_topics,
 )
+from raiden.constants import PATH_FINDING_BROADCASTING_ROOM
+# from matrix_client.errors import MatrixRequestError
```
Comment (palango) [inline src/pathfinding_service/service.py]: what about this

File: src/pathfinding_service/service.py [hunk 3]
```diff
@@ -100,16 +121,17 @@ def _setup_token_networks(self):
...
         self.token_network_registry_listener.start()
-
         self.is_running.wait()
 
     def stop(self):
         self.token_network_registry_listener.stop()
         for task in self.token_network_listeners:
             task.stop()
         self.is_running.set()
+        self.matrix_listener.stop()
```
Comment (palango) [inline src/pathfinding_service/service.py]: I think this should go before the  `self.is_running.set()`

File: src/pathfinding_service/service.py [hunk 4]
```diff
@@ -253,3 +275,13 @@ def create_token_network_for_address(
...
+
+    def handle_message(self, message: SignedMessage):
+        if isinstance(message, UpdatePFS):
+            self.on_pfs_update(message, UpdatePFS)
+            # TODO validate message
+        else:
+            log.info('Ignoring unknown message type')
+
+    def on_pfs_update(self, message, UpdatePFS):
+        print(f"das ist die message: {message}")
```
Comment (palango) [inline src/pathfinding_service/service.py]: no german please, and better make this a log message.

File: src/pathfinding_service/service.py [hunk 5]
```diff
@@ -253,3 +275,13 @@ def create_token_network_for_address(
...
         self.token_network_listeners.append(token_network_listener)
+
+    def handle_message(self, message: SignedMessage):
+        if isinstance(message, UpdatePFS):
+            self.on_pfs_update(message, UpdatePFS)
+            # TODO validate message
+        else:
+            log.info('Ignoring unknown message type')
+
+    def on_pfs_update(self, message, UpdatePFS):
```
Comment (palango) [inline src/pathfinding_service/service.py]: this should be `message: UpdatePFS`, right?

File: tests/pathfinding/conftest.py [hunk 6]
```diff
@@ -44,12 +44,17 @@ def no_greenlets_left():
     yield
     tasks = _get_running_greenlets()
     # give all tasks the chance to clean themselves up
+    for task in tasks:
```
Comment (palango) [inline tests/pathfinding/conftest.py]: Puh, this stuff is ugly

File: tests/pathfinding/test_blockchain_integration.py [hunk 7]
```diff
@@ -38,12 +38,13 @@ def test_pfs_with_mocked_client(
...
         required_confirmations=1,
         poll_interval=0,
+        private_key='3a1076bf45ab87712ad64ccb3b10217737f7faacbf2872e88fdd9a537d8fe266',
     )
 
     # greenlet needs to be started and context switched to
     pfs.start()
     wait_for_blocks(1)
-    gevent.sleep(0)
+    gevent.sleep(1)
```
Comment (palango) [inline tests/pathfinding/test_blockchain_integration.py]: Is this still necessary after the other changes?

File: tests/pathfinding/test_cli.py [hunk 8]
```diff
@@ -16,42 +16,54 @@
...
-def test_success():
+def test_success(default_cli_args):
     """ Calling the pathfinding_service with default args should succeed after heavy mocking """
     runner = CliRunner()
     with patch.multiple(**patch_args) as mocks:
         mocks['get_default_registry_and_start_block'].return_value = Mock(), Mock()
-        result = runner.invoke(main, [])
+        result = runner.invoke(
+            main,
+            default_cli_args + [],
```
Comment (palango) [inline tests/pathfinding/test_cli.py]: the `+ []` doesn't do anything

File: tests/pathfinding/test_rest.py [hunk 9]
```diff
@@ -64,6 +66,11 @@ def request_path_with(**kwargs):
     response = request_path_with(max_paths=-1)
     assert response.json()['errors'] == 'Number of paths must be positive: -1'
 
+    # killen aller greenlets
```
Comment (palango) [inline tests/pathfinding/test_rest.py]: english comments, and we have to find a better solution for this. I can have a look.

**Comment by palango** [comment]
Just a comment: this will also break the deployment. But this is no reason to not merge this.

File: src/pathfinding_service/cli.py (hunk 1 above)
Comment (Dominik1999) [reply src/pathfinding_service/cli.py]: I defined a `DEFAULT_POLL_INTERVALL = 10` in the pfs config

File: src/pathfinding_service/service.py (hunk 2 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: this was meant for the next debugging release. deleted it for now

File: src/pathfinding_service/service.py (hunk 4 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: changed to English, will be deleted soon. This is just for debugging

File: src/pathfinding_service/service.py (hunk 5 above)
Comment (Dominik1999) [reply src/pathfinding_service/service.py]: yes, changed it

File: tests/pathfinding/conftest.py (hunk 6 above)
Comment (Dominik1999) [reply tests/pathfinding/conftest.py]: yes it is, let's change it next week with Ulo :)

File: tests/pathfinding/test_cli.py (hunk 8 above)
Comment (Dominik1999) [reply tests/pathfinding/test_cli.py]: True, changed it

File: tests/pathfinding/test_blockchain_integration.py (hunk 7 above)
Comment (Dominik1999) [reply tests/pathfinding/test_blockchain_integration.py]: tests fail when it is removed. So it seems necessary. I suggest to keep it

File: src/raiden_libs/matrix.py [hunk 10]
```diff
@@ -56,12 +57,12 @@ def __init__(
 
     def _run(self):
         self.client.start_listener_thread()
-        self.client.sync_thread.get().wait()
```
Comment (palango) [inline src/raiden_libs/matrix.py]: Are you sure about this? I am not sure about the effect of this

Comment (Dominik1999) [reply src/raiden_libs/matrix.py]: Ehm it works like that. But I don't really know what it does either. I will put it on the open questions list

**Pull request merged**